### DISCUSSION TITLE
Case insensitive columns?

### DISCUSSION THREAD

**Discussion by Nv7-GitHub** [pocketbase/pocketbase#638] [Q&A]
In SQLite, case-insensitive columns exist. Is there a way to case-insensitively match strings using the `Dao`/make case-insensitive columns, or do I need to use the `DB` object to case-insensitively match strings?

**Comment by ganigeorgiev** [comment]
>  Is there a way to case-insensitively match strings using the Dao/make case-insensitive columns, or do I need to use the DB object to case-insensitively match strings?

I'm not sure that I understand the question. Could you provide an example what you are trying to do (pseudo code is OK)?

**Reply by Nv7-GitHub** [reply]
Lets say I have a field `name` (text). If a user makes something new with a name, I want to check if something already exists with that name. I want this check to be case-insensitive:

Lets say a record already exists with `name` "Hello", and a user suggests a name "hello" - I want this to still come up as it exists

SQLite supports a COLLATE NO CASE option to do this, and an option to make columns automatically do this, but I can't find out how to do this through the `Dao`. Is it possible?

**Reply by ganigeorgiev** [reply]
In this case you can construct a "raw" sql expression. For example:

```go
app.Dao().FindRecordsByExpr(collectiton, dbx.NewExp("name = {:name} COLLATE NO CASE", dbx.Params{"name": "test"}))
```

_But please note that this will work only for ASCII characters (you could also check https://github.com/pocketbase/pocketbase/issues/627)._
